images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -87,17 +96,38 @@ function Install-JavaFromAdoptOpenJDK {
     New-Item -ItemType File -Path $javaVersionPath -Name "$Architecture.complete" | Out-Null
 }
 
-$jdkVersions = (Get-ToolsetContent).java.versions
+$jdkVendors = (Get-ToolsetContent).java.vendors.name
+$defaultVendor = (Get-ToolsetContent).java.default_vendor
 $defaultVersion = (Get-ToolsetContent).java.default
 
-foreach ($jdkVersion in $jdkVersions) {
-    Install-JavaFromAdoptOpenJDK -JDKVersion $jdkVersion
+foreach ($jdkVendor in $jdkVendors) {
+
+    $jdkVersionsToInstall = ((Get-ToolsetContent).java.vendors | Where-Object { $_.name -eq $jdkVendor }).versions
```

Comment:
I agree with @al-cheb suggestion above.
Currently, you do:
```
$jdkVendors = (Get-ToolsetContent).java.vendors.name
foreach ($jdkVendor in $jdkVendors) {
  $jdkVersionsToInstall = ((Get-ToolsetContent).java.vendors | Where-Object { $_.name -eq $jdkVendor }).versions
...
```

I think we don't really need this filtering.
We can simplify logic to 
```
$jdkVendors = (Get-ToolsetContent).java.vendors
foreach ($jdkVendor in $jdkVendors) {
  $jdkVendor = $jdkVendor.name
  $jdkVersionsToInstall = $jdkVendor.versions
...
```

Reply:
@maxim-lobanov I have no really strong opinion on that suggestion, but I've glanced my drafts/backlogs and I remember that I was proposing something like you and @al-cheb do, but @miketimofeev suggested the current implementation as the more uniform one. Lets stick to the current one, after all it is just simple pipe-filtering with plain value substitution (and readability does not suffer, in my very opinion).

Reply:
@mikhailkoliada  I disagree to proceed with current implementation because it is unreasonably overcomplicated.

With your implementation, we iterate through array of objects and retrieve the name of all vendors. After that we iterate through array of names, trying to find object for every name and work with it. I don't really understand why we need to do it.
We should just iterate through the array of objects.